src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp

```diff
@@ -2240,7 +2240,14 @@ HRESULT CorProfiler::GenerateVoidILStartupMethod(const ModuleID module_id, mdMet
 #ifdef _WIN32
     WSTRING native_profiler_file = WStr("DATADOG.TRACE.CLRPROFILER.NATIVE.DLL");
 #else // _WIN32
-    WSTRING native_profiler_file = GetCoreCLRProfilerPath();
+    WSTRING native_profiler_file = GetEnvironmentValue(WStr("PROFID_{846F5F1C-F9AE-4B07-969E-05C26BC060D8}"));
```

Comment:
I asked before who reads `PROFID_...`, now I see it, but I do not fully understand. Is the path we expect there not the path of the Dll that will contain this code? Why do we need to pass it in this way? There is probably a very good reason, just want to understand.

Whatever the reason is, the env var is publically visible, i.e. it is an API. So we should name it better. I do not fully understand its purpose, so the following suggestion may or may not make sense, but perhaps something like:
`DD_TRACER_NATIVE_ENGINE_PATH` or `DD_INTERNAL_TRACER_NATIVE_ENGINE_PATH` to stress that this is not something people should play with
(and
`DD_PROFILER_NATIVE_ENGINE_PATH` \ `DD_INTERNAL_PROFILER_NATIVE_ENGINE_PATH`,
`DD_XYZPRODUCT_NATIVE_ENGINE_PATH` \ `DD_INTERNAL_XYZPRODUCT_NATIVE_ENGINE_PATH` for others).

Reply:
>I asked before who reads PROFID_..., now I see it, but I do not fully understand. Is the path we expect there not the path of the Dll that will contain this code? Why do we need to pass it in this way? There is probably a very good reason, just want to understand.

In the tracer, for non windows platform we use the absolute path to the native library as a PInvoke map to ensure it loads through the internal `dlopen`. A way to get this absolute path, is to read the same profiler variable reads by the runtime when loading the profiler.

Because now those variables point to the native loader profiler those PInvoke calls will fail (on non windows platforms). The workaround is to expose those path from the native loader with a well known key.

I can agree we can have a better naming for that variable (specially now that we differentiate the Profiler from the Tracer. It was generic before so we just concatenate a prefix with the profiler ID, each target profiler knows their ID). But  I'm not sure on using `Engine` in the name.

Following the .NET convention `CORECLR_PROFILER_PATH` and `COR_PROFILER_PATH`, I will change it to `DD_TRACER_PROFILER_PATH` and `DD_CONTINUOUSPROFILER_PROFILER_PATH`.

Reply:
done in 0df086a